src/main/java/com/minecolonies/coremod/colony/buildings/AbstractSchematicProvider.java

```diff
@@ -306,53 +289,25 @@ public boolean hasParent()
     @Override
     public void setParent(final BlockPos pos)
     {
-        if (!pos.equals(getID()))
+        if (isParentValid(pos))
         {
             parentSchematic = pos;
         }
     }
 
-    @Override
-    public Set<BlockPos> getChildren()
+    private boolean isParentValid(BlockPos position)
     {
-        // Validate childs existance
-        final IRegisteredStructureManager manager = colony.getBuildingManager();
-        List<BlockPos> toRemove = null;
-
-        for (final BlockPos pos : childSchematics)
-        {
-            if (manager.getBuilding(pos) == null)
-            {
-                if (toRemove == null)
-                {
-                    toRemove = new ArrayList<>();
-                }
-                toRemove.add(pos);
-            }
-        }
-
-        if (toRemove != null)
-        {
-            final Set<BlockPos> oldPositions = new HashSet<>(this.childSchematics);
-            oldPositions.removeAll(toRemove);
-            this.childSchematics = ImmutableSet.copyOf(oldPositions);
-        }
-
-        return childSchematics;
+        final IBuilding building = colony.getBuildingManager().getBuilding(position);
+        return building != null && !building.getID().equals(getID()) && !building.hasParent();
```

Comment:
Won't this block the case where a parent has a child that has another child?  That ought to still be valid.

Reply:
I asked Ray this and he basically said that wouldn't ever happen

Reply:
I don't think that's true.  It sounds like a reasonable thing to want to do.  It wouldn't even surprise me if a style like Byzantine already does it, since they have quite complex nesting (though I have not checked if they actually do).